src/object/object_domain.c

```diff
@@ -4994,6 +4995,154 @@ tp_atof (const DB_VALUE * src, double *num_value, DB_DATA_STATUS * data_stat)
   return status;
 }
 
+/*
+ * tp_str_to_vector - Coerce a string to a vector.
+ *    return: NO_ERROR or error code.
+ *    src(in): string DB_VALUE
+ *    result(out): vector DB_VALUE
+ * Note:
+ *    Accepts strings that are not null terminated. Don't call this unless
+ *    src is a string db_value.
+ */
+static int
+tp_str_to_vector (const DB_VALUE * src, DB_VALUE * result)
+{
+  const char *p = db_get_string (src);
+  const char *end = p + db_get_string_size (src);
+  int count = 0;
+  const int number_buffer_size = 64;
+  char number_buffer[number_buffer_size];
+  int buffer_idx;
+  const int max_vector_size = 2000;
+  float float_array[max_vector_size];
+  DB_SET *vec = NULL;
+  DB_VALUE e_val;
+  INTL_CODESET codeset = db_get_string_codeset (src);
+
+  if (p == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  // Skip leading spaces and opening bracket
+  p = (char *) intl_skip_spaces (p, end, codeset);
+  if (p >= end || *p != '[')
+    {
+      return ER_FAILED;
+    }
+  p++;
+
+  while (p < end && count < max_vector_size)
+    {
+      // Skip spaces before number
+      p = (char *) intl_skip_spaces (p, end, codeset);
+      if (p >= end)
+	{
+	  return ER_FAILED;
+	}
+
+      // Check for closing bracket
+      if (*p == ']')
+	{
+	  break;
+	}
+
+      // Get number into buffer
+      buffer_idx = 0;
+      while (p < end && *p != ',' && *p != ']' && buffer_idx < number_buffer_size)
+	{
+	  if (!isspace (*p))
+	    {
+	      number_buffer[buffer_idx++] = *p;
+	    }
+	  p++;
+	}
+
+      if (buffer_idx == 0 || buffer_idx >= number_buffer_size)
+	{
```

Comment:
```suggestion
      while (p < end && *p != ',' && *p != ']' && buffer_idx < number_buffer_size - 1)
	{
	  if (!isspace (*p))
	    {
	      number_buffer[buffer_idx++] = *p;
	    }
	  p++;
	}

      if (buffer_idx == 0 || buffer_idx >= number_buffer_size - 1)
	{
```

credits to @mhoh3963